src/chapters/forms/validation-with-javascript.rst

```diff
@@ -0,0 +1,140 @@
+               form.addEventListener("submit", function(event) {
+                  alert("submit clicked");
+               });
+            });
+         </script>
+         <body>
+            <form method="POST" action="https://www.w3schools.com/action_page.php">
+               <label>Username <input type="text" name="username"></label>
+               <label>Team Name <input type="text" name="team"></label>
+               <button>Submit</button>
+            </form>
+         </body>
+      </html>
+
+
+Follow Along as We Add Validation
+---------------------------------
+
+Use `this repl.it <https://repl.it/@launchcode/form-validation>`_ to add validation to
+the above example.
+
+To validate what the user has typed, we can get a reference to the ``input`` elements in
+the DOM and check ``input.value``. Let's change the *submit* event handler to alert
+the value of the username input. We are going to use
+``document.querySelector("input[name=username]")``, which uses an *attribute selector* to
+select the ``<input>`` that has ``name="username"``.
+
+.. sourcecode:: html
+   :linenos:
+
+   <script>
+      window.addEventListener("load", function() {
+         let form = document.querySelector("form");
+         form.addEventListener("submit", function(event) {
+            let usernameInput = document.querySelector("input[name=username]");
+            // alert the current value found in the username input
+            alert("username: " + usernameInput.value);
+         });
+      });
+   </script>
+
+Now that we know how to get the value of an input, we can add *conditional statements*.
+Let's add code that opens an alert box if *either* input values are *empty*.
+
+.. sourcecode:: html
+   :linenos:
+
+   <script>
+      window.addEventListener("load", function() {
+         let form = document.querySelector("form");
+         form.addEventListener("submit", function(event) {
+            let usernameInput = document.querySelector("input[name=username]");
+            let teamName = document.querySelector("input[name=team]");
+            if (usernameInput.value === "" || teamName.value === "") {
+               alert("All fields are required!");
+            }
+         });
+      });
+   </script>
+
+We are making progress. Now if you click *Submit* with one or both of the inputs empty,
+then an alert message appears telling you that both inputs are required.
+
+.. index:: ! preventDefault
+   single: event; preventDefault
+
+But we want to prevent the form submission from happening until all
+inputs have valid values. We can use the ``event`` parameter and
+``event.preventDefault()`` to stop the form submission. ``event.preventDefault()``
+prevents default browser functionality from happening, like form submission
+when ``<button>`` tags are clicked inside of a form. Remember that *event handler* functions
+are passed an ``event`` parameter which represents the occurring event.
```

Comment:
"...are passed an event argument which represents the event that the handler is responding to."